Add test for CreateDefaultDisk Setting

ttpcodes opened the pull request:
This PR adds a test for the `Setting: Create Default Disk on Labeled Nodes` that was implemented in longhorn/longhorn-manager#331. The test will check three different cases for when this `Setting` is enabled:
1. A `Disk` already exists on the `Node`. No additional `Disk` should be created.
2. A `Disk` doesn't exist on the `Node`, and the `Node` is labeled properly. The default `Disk` should be created at the path specified in `Setting: Default Data Path`.
3. A `Disk` doesn't exist on the `Node`, but the `Node` doesn't have the appropriate label. The default `Disk` should not be created.

Note that since additional permissions are needed to label the `Nodes` in this test, the `ClusterRole` created for the test has been modified accordingly.

This PR also contains a commit to set the `Namespace` for the test's `Resources` explicitly to `default` so that the test `Resources` are not inadvertently created in the wrong `Namespace`.

yasker commented on manager/integration/tests/test_node.py:
```diff
@@ -1045,3 +1084,79 @@ def test_replica_cleanup(client):  # NOQA
+    # Set disk creation and path Settings.
+    setting = client.by_id_setting(CREATE_DEFAULT_DISK_SETTING)
+    client.update(setting, value="true")
+    setting = client.by_id_setting(DEFAULT_DATA_PATH_SETTING)
```
I think you need to update DEFAULT_DATA_PATH_SETTING first, otherwise, the creation can happen as soon as you change the CREATE_DEFAULT_DISK_SETTING to true.

yasker commented on manager/integration/tests/test_node.py:
```diff
@@ -17,10 +21,45 @@
 
 
+@pytest.fixture
+def disk_path():
```
`random_disk_path`

ttpcodes commented:
- Switched the order of the calls to modify the `Default Data Path` and `Create Default Disk on Labeled Nodes`.
- Renamed `fixture` to `random_disk_path`.
